### PR TITLE
Update Button Guidance

### LINKED ISSUE #2082: Button Guidance - Implementing new button positioning guidance with examples
Implementation based on research and exploration from #2048, #2047, #1788, #1785

All items are linked to Figma designs.

Additional written guidance for these design pages:
- [ ] [Buttons](https://www.figma.com/file/Oi5UEEQ33VCAyOKQcZ8Nr8/HPE-Button-Component?node-id=4153%3A9026)
- [ ] [Layer](https://www.figma.com/file/Oi5UEEQ33VCAyOKQcZ8Nr8/HPE-Button-Component?node-id=4153%3A9007)
- [ ] [Form](https://www.figma.com/file/Oi5UEEQ33VCAyOKQcZ8Nr8/HPE-Button-Component?node-id=4153%3A8992)

Current examples to be changed:
- [ ] [Side Drawer Modal](https://www.figma.com/file/Oi5UEEQ33VCAyOKQcZ8Nr8/HPE-Button-Component?node-id=4153%3A9007)
- [ ] [Filter example in Forms page](https://www.figma.com/file/Oi5UEEQ33VCAyOKQcZ8Nr8/HPE-Button-Component?node-id=4153%3A8992)

### PR DESCRIPTION
**Pull request by taysea**
[Deploy Preview](https://deploy-preview-2110--keen-mayer-a86c8b.netlify.app/)

#### What does this PR do?

Updates Button guidance based on #2082 

#### Where should the reviewer start?
aries-site/src/pages/components/button.mdx

#### What testing has been done on this PR?

Local deploy.

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [x] Small, medium, and large screen sizes
- [x] Cross-browsers (FireFox, Chrome, and Safari)
- [x] Light & dark modes
- [x] All hyperlinks route properly

**Accessibility Checks**
- [x] Keyboard interactions
- [x] Screen reader experience
- [x] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [x] Console is free of warnings and errors
- [x] Passes E2E commit checks
- [x] Visual snapshots are reasonable

#### What are the relevant issues?
Closes #2082 

#### Should this PR be mentioned in Design System updates?
Yes.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

**Review by britt6612 — APPROVED** [review]
LGTM only thing is should we add a caption for the table?

File: aries-site/src/examples/components/button/ButtonAlignmentTable.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+export const ButtonAlignmentTable = () => (
+  <Card alignSelf="start" elevation="none" pad="medium" width="large">
+    {/* style tag is necessary in order for table not 
+    to exceed container width */}
+    <Table style={{ tableLayout: 'fixed', width: '100%' }}>
+      <TableHeader>
+        <TableRow>
+          {columns.map(c => (
+            <TableCell
+              border="bottom"
```
Comment (halocline) [inline aries-site/src/examples/components/button/ButtonAlignmentTable.js:76]: It feels like we are being inconsistent with our table styling. This styling differs from how they are applied on [Voice & Tone](https://design-system.hpe.design/foundation/voice-and-tone#other-common-problems-and-how-to-address-them) and our [DataTables](https://design-system.hpe.design/components/datatable).

I feel like we should be leaning on the theme here.

In addition, the [Figma file](https://www.figma.com/file/Oi5UEEQ33VCAyOKQcZ8Nr8/HPE-Button-Component?node-id=4153%3A9026) should probably also be using a Table component as opposed to text + lines to mimic a table, but isn't one...

There could be some rationale I am missing, but without that, I'm advocating for consistency and consuming the Design System's theme.

File: aries-site/src/examples/components/button/ButtonAlignmentTable.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                    {datum[c.property]}
+                  </Text>
+                ) : (
+                  <Box direction="row" gap="xsmall" wrap>
+                    {datum[c.property].map((item, index) => (
+                      <Fragment key={c.property}>
+                        {item.href ? (
+                          <Anchor
+                            label={item.label}
+                            href={item.href}
```
Comment (halocline) [inline aries-site/src/examples/components/button/ButtonAlignmentTable.js:104]: Are these going through the router? I believe we need Next Link.

File: aries-site/src/examples/components/layer/LayerSideDrawerExample.js [hunk 3]
```diff
@@ -97,11 +97,10 @@ const LayerForm = ({ setOpen }) => {
         <Box
           direction="row"
           gap="large"
-          justify="end"
           margin={{ top: 'medium', bottom: 'small' }}
         >
+          <Button label="Submit Form" primary type="submit" />
```
Comment (halocline) [inline aries-site/src/examples/components/layer/LayerSideDrawerExample.js:102]: The gap here is enormous. Is that what we want? Should be bumped down.

This is another instance where the Figma file doesn't seem to be using AutoLayout and proper spacing. The implementation in the file has a non-standard spacing of 41px.

Comment (halocline) [reply aries-site/src/examples/components/layer/LayerSideDrawerExample.js:102]: For comparison /templates/forms#filter is using medium and examples on https://design-system.hpe.design/templates/filtering?q=filt use small.

File: aries-site/src/examples/templates/forms/FilterExample.js [hunk 4]
```diff
@@ -105,6 +106,11 @@ export const FilterExample = () => {
               <CheckBoxGroup options={sellers} id="sellers" name="sellers" />
             </FormField>
           )}
+          <Box direction="row-responsive" gap="medium" pad={{ top: 'medium' }}>
```
Comment (halocline) [inline aries-site/src/examples/templates/forms/FilterExample.js:109]: Can we check on the responsive behavior here? Let's make sure it is aligned with the direction which @jeffstolz is heading. I believe the direction is calling for only primary buttons to occupy full-width. Let's also consider whether this scenario is covered in that guidance.

Screen Shot 2021-12-03 at 9 41 39 AM

File: aries-site/src/examples/components/button/ButtonAlignmentTable.js (hunk 1 above)
Comment (taysea) [reply aries-site/src/examples/components/button/ButtonAlignmentTable.js:76]: I am good with that an just falling back on what we get out of the box from the theme.

File: aries-site/src/examples/components/button/ButtonAlignmentTable.js (hunk 2 above)
Comment (taysea) [reply aries-site/src/examples/components/button/ButtonAlignmentTable.js:104]: Thank you, yes Next Link needed. Will update.

File: aries-site/src/examples/templates/forms/FilterExample.js (hunk 4 above)
Comment (taysea) [reply aries-site/src/examples/templates/forms/FilterExample.js:109]: Oh got it, I must have missed the primary only. I will take a look at his work there.

Comment (taysea) [reply aries-site/src/examples/templates/forms/FilterExample.js:109]: Followed up with Jeff -- he is going to revise his guidance to clarify, but he thinks that on mobile buttons should go full-width in these contexts and color/styling provides correct indication of hierarchy.

**Review by halocline — APPROVED** [review]

**Pull request merged**